test/database/openshiftclusters.go

```diff
@@ -0,0 +1,173 @@
+package database
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/fvbommel/sortorder"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+)
+
+type ByKey []*api.OpenShiftClusterDocument
+
+func (a ByKey) Len() int           { return len(a) }
+func (a ByKey) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a ByKey) Less(i, j int) bool { return sortorder.NaturalLess(a[i].Key, a[j].Key) }
+
+func getQueuedOpenShiftDocuments(client cosmosdb.OpenShiftClusterDocumentClient) (res []*api.OpenShiftClusterDocument) {
+	docs := fakeOpenShiftClustersGetAllDocuments(client)
+
+	for _, r := range docs {
+		var include bool
+
+		switch r.OpenShiftCluster.Properties.ProvisioningState {
+		case
+			api.ProvisioningStateCreating,
+			api.ProvisioningStateUpdating,
+			api.ProvisioningStateAdminUpdating,
+			api.ProvisioningStateDeleting:
+			include = true
+		}
+
+		if include && (r.LeaseExpires > 0 && int64(r.LeaseExpires) < time.Now().Unix()) {
+			include = false
+		}
+
+		if include {
+			res = append(res, r)
+		}
+	}
+	return
+}
+
+func fakeOpenShiftClustersQueueLengthQuery(client cosmosdb.OpenShiftClusterDocumentClient, query *cosmosdb.Query, options *cosmosdb.Options) cosmosdb.OpenShiftClusterDocumentRawIterator {
```

Comment:
I'm worried that this stuff is a long way away from its equivalent in pkg/database - should this be pkg/database/test for example?

Reply:
we can always move it later.